content/posts/xapi-activities-101/index.md

```diff
@@ -0,0 +1,262 @@
+
+## Shopping for activities 🛒
+
+### Activity definitions
+
+Activity definitions are found in multiple places and can be merged in different ways according both
+to the LRS spec and implementation.
+
+#### In xAPI statements
+
+This is the most common place for activity definition.  
+Activity definitions are found in two places inside a statement :
+
+**📌 In the `object` property**
+
+Only
+[when the ObjectType is an Activity](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#2441-when-the-objecttype-is-activity),
+(actually the most common case with statements) the activity definition is found in the `definition`
+property of the `object`. This means `statement.object.definition` is an activity definition (big up
+if you deducted that yourself 😉).
+
+**📌 In the `context` property**
+
+The statement optional `context` can have a `contextActivities` property that is
+[_A map of the types of learning activity context that this Statement is related to. Valid context types are: "parent", "grouping", "category" and "other"_](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#2462-contextactivities-property:~:text=A%20map%20of%20the%20types%20of%20learning%20activity%20context%20that%20this%20Statement%20is%20related%20to.%20Valid%20context%20types%20are%3A%20parent%2C%20%22grouping%22%2C%20%22category%22%20and%20%22other%22)
+
+According to the spec : _Many Statements do not just involve one (Object) Activity that is the
+focus, but relate to other contextually relevant Activities. The "contextActivities" property allows
+for these related Activities to be represented in a structured manner._
+
+Statements matching these activities can be queries via a GET on `/xAPI/statements` by setting the
+query parameter
+[`related_activities`](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Communication.md#:~:text=Apply%20the%20Activity,the%20%22activity%22%20parameter.)
+to True. The response will _include statements for which the Object, any of the context Activities,
+or any of those properties in a contained SubStatement match the Activity parameter._
+
+That was a mouthful, let's digest it with a strong double espresso ☕ before we follow on.
+
+### [Activities resource](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Communication.md#22-document-resources:~:text=The%20Activities%20Resource%20provides%20a%20method%20to%20retrieve%20a%20full%20description%20of%20an%20Activity%20from%20the%20LRS)
+
+As well as the statements resource, the xAPI spec defines other
+[resources](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Communication.md#20-resources)
+including the activities resource. _The Activities Resource provides a method to retrieve a full
+description of an Activity from the LRS._
+
+The request will return the **canonical activity definition** 👾 of the LRS. We'll explain this
+concept [below](#canonical-activity-definition-).
```

Comment:
Can you provide an example request/response to the LRS?

Reply:
I tried

For example, executing an HTTP GET request to `/activities`  : `curl localhost/activities?activityId=http://funcourse`
could return

```json
{
    "id": "http://funcourse",
    "definition": {
        "type": "http://adlnet.gov/expapi/activities/course",
        "name": {
            "en-US": "FUN-f1bdf5a4-8b9e-582b-aa22-2d7e3a6e434a"
        }
    },
    "objectType": "Activity"
}
```